lib/snyk/package.go

```diff
@@ -29,6 +30,51 @@ import (
 const snykServer = "https://api.snyk.io/rest"
 const version = "2023-04-28"
 
+func purlToSnykURL(basePath string, purl packageurl.PackageURL, nsDelimiter string) string {
+	sb := strings.Builder{}
+	sb.WriteString(basePath)
+	if purl.Namespace != "" {
+		sb.WriteString(purl.Namespace)
+		sb.WriteString(nsDelimiter)
+	}
+	sb.WriteString(purl.Name)
+	return sb.String()
+}
```

Comment:
It took me multiple times of reading to understand how this funtion interacts with callees and what it does. I feel like this abstraction is complicating things, and we don't actually need a `strings.Builder`. Can we sacrifice this for a bit more duplication and rather construct strings in the functions that call this?